python/lsst/daf/ingest/index_exposure_task.py

```diff
@@ -0,0 +1,358 @@
+                # See if there is already an entry for the given data id.
+                self.cursor.execute(
+                    'SELECT rowId FROM {}\n'
+                    '    WHERE pickledDataId = ?'.format(table_name),
+                    (pickled_data_id,)
+                )
+                results = self.cursor.fetchall()
+                if len(results) > 0:
+                    # If so, update spatial information for the exposure.
+                    row_id = results[0][0]
+                    self.cursor.execute(
+                        'UPDATE {}\n'
+                        '    SET encodedPolygon = ?\n'
+                        '    WHERE rowId = ?'.format(table_name),
+                        (encoded_polygon, row_id)
+                    )
+                    self.cursor.execute(
+                        'UPDATE {} SET\n'
+                        '    xMin = ?, xMax = ?,\n'
+                        '    yMin = ?, yMax = ?,\n'
+                        '    zMin = ?, zMax = ?\n'
+                        'WHERE rowId = ?'.format(index_table_name),
+                        (bbox.x().getA(), bbox.x().getB(),
+                         bbox.y().getA(), bbox.y().getB(),
+                         bbox.z().getA(), bbox.z().getB(),
+                         row_id)
+                    )
+                    return
+            # Insert the data id and corresponding spatial information.
+            self.cursor.execute(
+                'INSERT INTO {}\n'
+                '    (pickledDataId, encodedPolygon)\n'
+                '    VALUES (?, ?)'.format(table_name),
+                (pickled_data_id, encoded_polygon)
+            )
+            row_id = self.cursor.lastrowid
+            self.cursor.execute(
+                'INSERT INTO {}\n'
+                '     (rowId, xMin, xMax, yMin, yMax, zMin, zMax)\n'
+                '     VALUES (?, ?, ?, ?, ?, ?, ?)'.format(index_table_name),
+                (row_id,
+                 bbox.x().getA(), bbox.x().getB(),
+                 bbox.y().getA(), bbox.y().getB(),
+                 bbox.z().getA(), bbox.z().getB())
+            )
+
+    def _create_tables(self, connection, table_name, index_table_name):
+        """Create SQLite 3 tables for the index.
+
+        One table is for exposure data ids and boundaries, and the other is
+        an `R*Tree`_ of exposure bounding boxes.
+
+        .. _`R*Tree`:      https://www.sqlite.org/rtree.html
+        """
+        for statement in self.config.init_statements:
+            connection.execute(statement)
+        connection.execute(
+            'CREATE VIRTUAL TABLE IF NOT EXISTS {} USING rtree(\n'
+            '    rowId,\n'
+            '    xMin, xMax,\n'
+            '    yMin, yMax,\n'
+            '    zMin, zMax\n'
+            ')'.format(index_table_name)
+        )
+        connection.execute(
+            'CREATE TABLE IF NOT EXISTS {} (\n'
+            '    rowId INTEGER PRIMARY KEY,\n'
```

Comment:
See https://www.sqlite.org/autoinc.html : you get that automatically just by omitting a rowId value when you INSERT. The only thing AUTOINCREMENT gives you in addition is that it prevents reuse of ids from previously deleted rows.